Result/ResultType.swift

```diff
@@ -51,6 +51,18 @@ public extension ResultType {
 			ifFailure: Result<U, Error>.Failure)
 	}
 	
+	/// Returns a new Result by mapping `Success`es’ values using `transform`, or wrapping errors thrown as `Failure`s’.
+	public func tryMap<U>(@noescape transform: Value throws -> U) -> Result<U, Error> {
+        return flatMap { value in
+            do {
+                return .Success(try transform(value))
+            }
+            catch {
+                return .Failure(error as! Error)
```

Comment:
This is about half of what’s wrong with Swift’s error handling semantics in a nutshell 😞

Would it be better to add a constrained extension such that `Error` types which can be constructed with an arbitrary `ErrorType` get `tryMap` and others do not?